### DISCUSSION TITLE
Timed out & What does /XX mean?

### DISCUSSION THREAD

**Discussion by foxypiratecove37350** [touhidurrr/iplist-youtube#12] [Q&A] [answered]
Each time I try to access to one of the IP addresses, I get a Timed Out error.
An I'm wondering what does the `/XX` mean in the IP addresses?

Thanks.

**Comment by touhidurrr** [comment, marked as answer]
If you want to ping IPs then use [ipv4_list.txt](https://github.com/touhidurrr/iplist-youtube/blob/main/ipv4_list.txt) or [ipv6_list.txt](https://github.com/touhidurrr/iplist-youtube/blob/main/ipv6_list.txt), not the CIDR lists.

**IP/bits** is format known as CIDR notation. As you know, IPv4 is 32 bits and IPv6 is 128 bits. (You can kinda write a single IP also in this notation for example `192.168.1.2/32` but that is not the intention here) The bits in this notation determine how many bits from left to right will remain unchanged in a subnet. So if I say for example `192.168.1.0/24`, this means the first 24 bits remains unchanged in this subnet (which is `192.168.1`) and rest of the bits can change. So, this basically means the IPs `192.168.1.0` to `192.168.1.255` are all part of this subnet. Similarly, `192.168.0.0/16` will mean a subnet of `192.168.0.0` to `192.168.255.255`.


That's my utmost attempt to describe this. To know more see: https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing#CIDR_notation

Note that CIDR notation is not used to describe a single IP but a group of IPs sequentially. I am generating lists in this format because it is useful for routing. (You can if you want to, but that's not the intention here) Also because IPs (especially IPv4) are assigned and sold in 8-bit or 10-bit blocks so any company having some IPs will have ownership of IPs grouped in at least an 8-bit block not just randomly, so, the script generating them kind of assume that if you own an IP, you at least own its 8-bit block. This is an assumption that I made up. For IPv6, They are assigned in at least 80-bit blocks, but the script assumes it is 64. (I think I will change it if someone complains)

**Comment by foxypiratecove37350** [comment]
Thanks for explaining.

I tried to ping some IPs of the [`ipv4_list.txt`](https://github.com/touhidurrr/iplist-youtube/blob/main/ipv4_list.txt) and [`ipv6_list.txt`](https://github.com/touhidurrr/iplist-youtube/blob/main/ipv6_list.txt) and I'm still getting a timed out, but anyway I won't use it because it's really not durable in time.

Still thanks.

**Comment by touhidurrr** [comment]
> I tried to ping some IPs of the [ipv4_list.txt](https://github.com/touhidurrr/iplist-youtube/blob/main/ipv4_list.txt) and [ipv6_list.txt](https://github.com/touhidurrr/iplist-youtube/blob/main/ipv6_list.txt) and I'm still getting a timed out, but anyway I won't use it because it's really not durable in time.

I tried some random IPs and many of them did time out.  Being in the list means YouTube used the IP at some point of time or it was returned from their DNS servers at least but it does not mean that they are using it now or they are online. This is not the purpose of the lists either.

If you are trying to find active IPs then maybe randomly select some IPs and ping them.
